Sources/Misc/Signing+ResponseVerification.swift

```diff
@@ -90,31 +112,18 @@ extension Result where Success == Data?, Failure == NetworkError {
         verificationMode: Signing.ResponseVerificationMode
     ) -> Result<HTTPResponse<Data?>, Failure> {
         return self.flatMap { body in
-            if let body = body {
-                let response = HTTPResponse.create(
-                    with: response,
-                    body: body,
-                    request: request,
-                    publicKey: verificationMode.publicKey,
-                    signing: signing
-                )
-
-                if response.verificationResult == .failed, case .enforced = verificationMode {
-                    return .failure(.signatureVerificationFailed(path: request.path))
-                } else {
-                    return .success(response.mapBody(Optional.some))
-                }
+            let response = HTTPResponse.create(
+                with: response,
+                body: body ?? .init(),
```

Comment:
Hmm I wonder if we should make the `HTTPResponse.body` parameter nullable instead of setting to an empty `Data`... We would have to guard against that later on but seems like a better representation? NABD though.

Reply:
`HTTPResponse` is generic over its `Body` type.
Initially we do create an `HTTPResponse<Data?>`, and this method returns that, but ``Signing.verify` takes a body as `Data`. I think it's simpler to deal with only one valid representation of an empty body.
This is the same thing with `[]`, `{}, `""`. Ideally the language allows a way to represent in the type system a value that can't be empty, but since we can't, I think it's better to make the parameter to `Signing.verify` non-optional so we don't have to handle both cases.